support-frontend/assets/helpers/paymentIntegrations/readerRevenueApis.js

```diff
@@ -40,27 +40,37 @@ export type StripePaymentMethod = 'StripeCheckout' | 'StripeApplePay' | 'StripeP
 export type StripePaymentRequestButtonMethod = 'none' | StripePaymentMethod;
 
 type RegularContribution = {|
+  productType: 'Contribution',
   amount: number,
   currency: string,
   billingPeriod: BillingPeriod,
 |};
 
 export type DigitalSubscription = {|
+  productType: 'DigitalSubscription',
   currency: string,
   billingPeriod: BillingPeriod,
   readerType: ReaderType,
-  giftMessage?: string | null,
-  giftDeliveryDate?: string | null,
 |};
 
 export type PaperSubscription = {|
+  productType: 'Paper',
   currency: string,
   billingPeriod: BillingPeriod,
   fulfilmentOptions: FulfilmentOptions,
   productOptions: ProductOptions,
 |};
 
-type ProductFields = RegularContribution | DigitalSubscription | PaperSubscription
+export type WeeklySubscription = {|
+  productType: 'GuardianWeekly',
+  currency: string,
+  billingPeriod: BillingPeriod,
+  fulfilmentOptions: FulfilmentOptions,
+|};
+
+export type SubscriptionProductFields = DigitalSubscription | PaperSubscription | WeeklySubscription
```

Comment:
not sure if to add this here or where it's used?  It was only needed because putting it inline was causing a line longer than 120 characters that couldn't be split.

Reply:
Fine to put it here 👍